src/Simplex/Messaging/Crypto.hs

```diff
@@ -1384,3 +1405,9 @@ keyError :: (a, [ASN1]) -> Either String b
 keyError = \case
   (_, []) -> Left "unknown key algorithm"
   _ -> Left "more than one key"
+
+x509ToPrivateSign :: PrivKey -> Either String APrivateSignKey
```

Comment:
this seems unnecessary, there is already x509ToPrivate, it can be combined with type check

Reply:
with CryptoPrivateKey instance